pkgs/shells/zsh/zplugin/default.nix

```
@@ -0,0 +1,40 @@
+{ stdenvNoCC, lib, fetchFromGitHub }:
+stdenvNoCC.mkDerivation rec {
+  pname = "zplugin";
+  version = "2.3";
+  src = fetchFromGitHub {
+    owner = "zdharma";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "0qqv5p19s8jb06d6h55dm4acji9x2rpxb2ni3h7fb0q43iz6y85w";
+  };
+  # adapted from https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=zsh-zplugin-git
+  dontBuild = true;
+  installPhase = ''
+    outdir="$out/share/$pname"
+
+    cd "$src"
+
+    # Zplugin's source files
+    install -dm0755 "$outdir"
+    install -m0644 zplugin{,-side,-install,-autoload}.zsh "$outdir"
+    install -m0755 git-process-output.zsh "$outdir"
+
+    # Zplugin autocompletion
+    install -dm0755 "$out/share/zsh/site-functions"
+    install -m0744 _zplugin "$out/share/zsh/site-functions"
```

can be replaced with 
```suggestion
    installShellCompletion --zsh _zplugin
```